google/cloud/bigtable/batcher.py

```diff
@@ -51,36 +53,54 @@ def __init__(self, max_mutation_bytes=MAX_MUTATION_SIZE, flush_count=FLUSH_COUNT
         self.total_size = 0
         self.max_mutation_bytes = max_mutation_bytes
         self.flush_count = flush_count
+        self._lock = threading.Lock()
 
     def get(self):
-        """Retrieve an item from the queue. Recalculate queue size."""
-        row = self._queue.get()
-        mutation_size = row.get_mutations_size()
-        self.total_mutation_count -= len(row._get_mutations())
-        self.total_size -= mutation_size
-        return row
+        """
+        Retrieve an item from the queue. Recalculate queue size.
+
+        If the queue is empty, return None.
+        """
+        with self._lock:
+            try:
+                row = self._queue.get_nowait()
+                mutation_size = row.get_mutations_size()
+                self.total_mutation_count -= len(row._get_mutations())
+                self.total_size -= mutation_size
+                return row
+            except queue.Empty:
+                return None
+
+    def get_all(self):
+        """Get all items from the queue."""
+        with self._lock:
+            items = []
+            while not self._queue.empty():
+                items.append(self._queue.get_nowait())
+            self.total_mutation_count = 0
+            self.total_size = 0
+            return items
 
     def put(self, item):
         """Insert an item to the queue. Recalculate queue size."""
 
         mutation_count = len(item._get_mutations())
 
-        self._queue.put(item)
+        with self._lock:
+            self._queue.put(item)
 
-        self.total_size += item.get_mutations_size()
-        self.total_mutation_count += mutation_count
+            self.total_size += item.get_mutations_size()
```

Comment:
same here, can these 2 lines be moved out of the lock?

Reply:
I removed the lock, let me know what you think